packages/graphql-hooks/src/useSubscription.ts

```diff
@@ -50,8 +62,35 @@ function useSubscription<
     return () => {
       subscription.unsubscribe()
     }
-  }, []) // eslint-disable-line
-  // the effect should be run when component is mounted and unmounted
+  }, [options.query, variablesRef.current])
 }
 
 export default useSubscription
+
+/**
+ * Compares two objects for equality _only considering the first level of object keys_.
+ * @example
+ * var o1 = {a: 1, b: 2}
+ * var o2 = {a: 1, b: 2}
+ * isEqual(o1, o2) === true
+ *
+ * var o1 = {a: 2, b: 4}
+ * var o2 = {a: 9, b: 'fish', c: 'why is this here?'}
+ * isEqual(o1, o2) === false
+ */
+function isEqual(o1: object, o2: object): boolean {
```

Comment:
I'd suggest moving this function in the `packages/graphql-hooks/src/utils.ts` file.